src/modules/roc_sndio/target_sox/roc_sndio/sox_sink.cpp

```diff
@@ -137,7 +137,8 @@ void SoxSink::write(audio::Frame& frame) {
 }
 
 bool SoxSink::setup_buffer_() {
-    buffer_size_ = packet::ns_to_size(frame_length_, sample_rate(), channels_);
+    audio::SampleSpec ss = audio::SampleSpec(sample_rate(), sample_spec_.getChannels());
+    buffer_size_ = ss.ns_to_size(frame_length_);
```

Comment:
Here, we use sample_rate() to get up-to-date sample rate from `output_->signal`, which is updated in open().

Instead of constructing temporary SampleSpec here, let's update sample_spec_.rate in open() (set it to `output_->signal.rate`), so that we'll have correct sample_spec_ after open() and can use it here.

Reply:
Same for SoxSource::setup_buffer_()